code/API_definitions/EdgeCloud_LcM.yaml

```diff
@@ -0,0 +1,658 @@
+ 
+security:
+  - oAuth2ClientCredentials: 
+    - nbi-mgmt
+ 
+tags:
+  - name: Applications
+    description: Manage of Applications
+  - name: Application Artifacts 
+    description: Manage of Artifacts
+  - name: Mobile Edge Cloud Platform
+    description: Mobile Edge Nodes Available
+ 
+paths:
+  /app:
+    post: 
+      tags:
+        - Applications
+      summary: Provision an application 
+      description: Ask the operator to provision an application to one or several Edge Application Servers taking into account resources (e.g. vCPU, Memory, network interfaces, storage, GPU) 
+      operationId: appProvisioning
+      requestBody:
+        description: Details about application and zones where application instance should be instantiated
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/AppProvisioning'
+        required: true      
+      responses:
+        '201':
+          description: Application provisioned
+          content: 
+            application/json:
+              schema:
+                type: object
+                properties:
+                  appId:
+                    $ref: '#/components/schemas/AppId'
+                  aplicationInstace:
+                    type: array
+                    items:
+                      $ref:  '#/components/schemas/ProvisionedApp'
+                minItems: 1                 
+        '400':
+          $ref: '#/components/responses/400'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '409':
+          description: Conflict
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorInfo'
+              example:
+                status: 409
+                code: CONFLICT
+                message: "Application already provisioned"
+        '500':
+          $ref: '#/components/responses/500'
+        '501':
+          $ref: '#/components/responses/501'
+        '503':
+          $ref: '#/components/responses/503'
+ 
+  /app/{appId}:
+    delete:
+      tags:
+        - Applications
+      summary: Terminate an Application
+      description: Terminate the running instance of an application in one or several Edge Application Servers
+      operationId: deleteApplication
+      parameters:
+        - name: appId
+          in: query
+          description: Identificator of the application generated by the OP NBI once the provisioning was successful, terminate all the instances related to this appId
+          required: false
+          schema:
+            $ref: "#/components/schemas/AppId"
+        - name: appInstanceId
+          in: query
+          description: Identificator of the specific application instance that will be terminated
+          required: false
+          schema:
+            $ref: "#/components/schemas/AppInstanceId"
```

Comment:
It was defined this way with this rationale:
-  If a user would delete an appId they will be deleting all the appInstances asociated to it
- If a user wanted to delete only a appInstance, they could make the delete request including app instance in the request body

If finally artifact and app are unified as a same entity this could also simplifyed. So we suggest to wait until that discussion is resolved.